assets/src/js/admin/stripe-admin.js

```diff
@@ -232,6 +233,138 @@ window.addEventListener( 'DOMContentLoaded', function() {
+            <input type="text" minlength="5" maxlength="22">
+            <button class="button-primary" disabled>${__('Save', 'give')}</button>
+            <button class="button-secondary">${__('Cancel', 'give')}</button>`;
+
+        editStripeStatementDescriptor.forEach((actionLink) => {
+            actionLink.addEventListener(
+                'click',
+                (e) => {
+                    e.preventDefault();
+
+                    let container = actionLink.closest('.give-stripe-connect-data-field'),
+                        containerDisplayStylePropertyValue = container.style.display;
+
+                    // Enable statement descriptor editing mode.
+                    container.insertAdjacentHTML('afterend', formTemplate);
+                    container.style.display = 'none';
+
+                    let inputField = container.nextElementSibling,
+                        saveButton = inputField.nextElementSibling,
+                        cancelButton = saveButton.nextElementSibling;
+
+                    // Add style.
+                    inputField.value = getStripeStatementDescriptorText();
+                    inputField.style.display = 'block';
+                    inputField.style.marginBottom = '10px';
+                    saveButton.style.marginRight = '5px';
+
+                    // Add events.
+                    inputField.addEventListener(
+                        'keyup',
+                        (e) => {
+                            let newStatementDescriptor = inputField.value.trim();
+
+                            if (!newStatementDescriptor || newStatementDescriptor === getStripeStatementDescriptorText()) {
+                                saveButton.disabled = true;
+                                return;
+                            }
+
+                            saveButton.disabled = false;
+                        }
+                    );
+
+                    cancelButton.addEventListener(
+                        'click',
+                        (e) => {
+                            e.preventDefault();
+                            exitStatementDescriptorEditingMode()
+                        });
+
+                    saveButton.addEventListener(
+                        'click',
+                        (e) => {
+                            e.preventDefault();
+
+                            let newStatementDescriptorText = getNewStatementDescriptor();
+                            let actionUrl = `${container.getAttribute('data-action-url')}&statement-descriptor=${encodeURIComponent(newStatementDescriptorText)}`
+
+                            if( ! isValidaStatementDescriptor(newStatementDescriptorText) ) {
+                                new Give.modal.GiveErrorAlert({
+                                    modalContent:{
+                                        title: __( 'Invalid Statement Descriptor Text', 'give'),
+                                        desc: sprintf(
```

Comment:
@Benunc Can you provide feedback on this error message? This message will appear in a modal.

Reply:
@ravinder what could the potential causes of this error be? From a brief glance, it looks like it's either too short/too long, or it has a special character or something. How can we be more specific here, and what it the next step for the user?